src/main/java/com/spotify/reaper/resources/CommonTools.java

```diff
@@ -133,78 +142,47 @@ public static RepairRun registerRepairRun(AppContext context, Cluster cluster,
-      throws ReaperException {
-    RepairRun.Builder runBuilder = new RepairRun.Builder(cluster.getName(), repairUnit.getId(),
-                                                         DateTime.now(), intensity,
-                                                         segments, repairParallelism);
-    runBuilder.cause(cause.isPresent() ? cause.get() : "no cause specified");
-    runBuilder.owner(owner);
-    RepairRun newRepairRun = context.storage.addRepairRun(runBuilder);
-    if (newRepairRun == null) {
-      String errMsg = String.format("failed storing repair run for cluster \"%s\", "
-                                    + "keyspace \"%s\", and column families: %s",
-                                    cluster.getName(), repairUnit.getKeyspaceName(),
-                                    repairUnit.getColumnFamilies());
-      LOG.error(errMsg);
-      throw new ReaperException(errMsg);
-    }
-    return newRepairRun;
+  private static RepairRun.Builder createNewRepairRun(
+          Cluster cluster,
+          RepairUnit repairUnit,
+          Optional<String> cause,
+          String owner,
+          int segments,
+          RepairParallelism repairParallelism,
+          Double intensity) throws ReaperException {
+
+    return new RepairRun.Builder(cluster.getName(), repairUnit.getId(), DateTime.now(), intensity, segments, repairParallelism)
+            .cause(cause.isPresent() ? cause.get() : "no cause specified")
+            .owner(owner);
   }
 
   /**
    * Creates the repair runs linked to given RepairRun and stores them directly in the storage
    * backend.
    */
-  private static RepairRun storeNewRepairSegments(AppContext context, List<RingRange> tokenSegments,
-                                             RepairRun repairRun, RepairUnit repairUnit) {
+  private static List<RepairSegment.Builder> createRepairSegments(List<RingRange> tokenSegments, RepairUnit repairUnit){
+
     List<RepairSegment.Builder> repairSegmentBuilders = Lists.newArrayList();
-    for (RingRange range : tokenSegments) {
-      RepairSegment.Builder repairSegment = new RepairSegment.Builder(repairRun.getId(), range,
-                                                                      repairUnit.getId());
-      repairSegmentBuilders.add(repairSegment);
-    }
-    context.storage.addRepairSegments(repairSegmentBuilders, repairRun.getId());
-    if (repairRun.getSegmentCount() != tokenSegments.size()) {
-      LOG.debug("created segment amount differs from expected default {} != {}",
-                repairRun.getSegmentCount(), tokenSegments.size());
-      RepairRun newRepairRun = repairRun.with().segmentCount(tokenSegments.size()).build(repairRun.getId());
-      context.storage.updateRepairRun(newRepairRun);
-      
-      return newRepairRun;
-    }
-    
-    return repairRun;
+    tokenSegments.forEach((range) -> repairSegmentBuilders.add(new RepairSegment.Builder(range, repairUnit.getId())));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the parentheses around the "range" parameter [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1611)